### PR TITLE
Require CAS authentication for all pages 

### PR DESCRIPTION
**Pull request by bess**
* Protect all routes in the management application behind CAS authentication.
* Update tests to match this expectation.

Connected to https://github.com/yalelibrary/YUL-DC/issues/602

Co-authored-by: @maxkadel 
Co-authored-by: @orangewolf

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -37,6 +37,4 @@
   authenticated :user do
     mount DelayedJobWeb, at: "/delayed_job"
   end
-  # fall back if not authenticated
-  get '/delayed_job', to: redirect('/management/users/sign_in')
```
Comment (orangewolf) [inline config/routes.rb:41]: I am pretty sure this should be `get '/delayed_job', to: redirect('/management/users/auth/cas')`

**Review by orangewolf — DISMISSED** [review]
Small change to fix in delayed_job login exception

File: config/routes.rb (hunk 1 above)
Comment (bess) [reply config/routes.rb:41]: Fixed

**Comment by bess** [comment]
@orangewolf Is this what you meant when you said we should raise an exception to prevent a redirect loop? 

```
    def after_omniauth_failure_path_for(_resource)
      raise 'CAS Authentication is not working'
    end

```

**Review by maxkadel — APPROVED** [review]

**Comment by bess** [comment]
When we tried to deploy this branch we realized we need a place outside of CAS that the health checks can look at to ensure the application is running. After conversation with @mark-dce we decided to expose part of the management#index page, just enough that it prompts you to log in if you go to the root path of the application without authentication: 

Screen Shot 2020-09-24 at 11 27 18 AM

**Review by maxkadel — APPROVED** [review]

**Pull request merged**